### PR TITLE
add cls command to cli

### PR DESCRIPTION
**Pull request by lukasheinrich**
# Description

addresses #204 

both the pipe and the read from file work

# Checklist Before Requesting Approver

- [x] Tests are passing
- [x] "WIP" removed from the title of the pull request

### CODE REVIEW

File: pyhf/commandline.py [hunk 1]
```diff
@@ -25,3 +27,18 @@ def xml2json(entrypoint_xml, basedir, output_file, track_progress):
...
         log.info("Written to {0:s}".format(output_file))
+
+@pyhf.command()
+@click.argument('workspace', default = '-')
+@click.option('--output-file', help='The location of the output json file. If not specified, prints to screen.', default=None)
+def cls(workspace,output_file):
+    specstream = click.open_file(workspace)
+    d = json.load(specstream)
+    p = Model({'channels':d['channels']}, poiname=d['toplvl']['measurements'][0]['config']['poi'])
+    result = runOnePoint(1.0, d['data']['channel1'] + p.config.auxdata, p)
```
Comment (lukasheinrich) [inline pyhf/commandline.py]: this is hardcoded to take the 'first' measurement. in my experience not a lot of analyses actually use the measurement feature all that much..99% probably only have one

File: pyhf/commandline.py [hunk 2]
```diff
@@ -25,3 +27,18 @@ def xml2json(entrypoint_xml, basedir, output_file, track_progress):
     else:
         json.dump(spec, open(output_file, 'w+'), indent=4, sort_keys=True)
         log.info("Written to {0:s}".format(output_file))
+
+@pyhf.command()
+@click.argument('workspace', default = '-')
+@click.option('--output-file', help='The location of the output json file. If not specified, prints to screen.', default=None)
+def cls(workspace,output_file):
+    specstream = click.open_file(workspace)
```
Comment (lukasheinrich) [inline pyhf/commandline.py]: not sure how to check for existence in case of non `-` in a click-y way

**Comment by lukasheinrich** [comment]
to make this work #200 must be merged (includes some fixes)

**Comment by lukasheinrich** [comment]
https://asciinema.org/a/lyQ8xsxCfeSxeirSQlNSIoU4G

**Comment by lukasheinrich** [comment]
this also works

```
cat recast.json|pyhf cls workspace.json --patch -
cat recast.json|pyhf cls <(curl -s http://physics.nyu.edu/~lh1132/workspace.json) --patch -
```

File: pyhf/commandline.py (hunk 2 above)
Comment (kratsg) [reply pyhf/commandline.py]: Linux-y would also allow someone to pipe-in the JSON output into this function, no? `pyhf xml2json .... | pyhf cls`

File: pyhf/commandline.py (hunk 1 above)
Comment (kratsg) [reply pyhf/commandline.py]: Should spit a warning if more than one `poiname` detected -- and also do some logging to indicate which one is being used.

File: pyhf/commandline.py (hunk 2 above)
Comment (lukasheinrich) [reply pyhf/commandline.py]: this should already work. my point is that for it I had to remove the `type = click.Path`

Comment (lukasheinrich) [reply pyhf/commandline.py]: the reason it works is that click is smart about `-` so wherever it is '-' the stdin (if mode 'r') or stdout (if mode 'w') is written to

**Comment by kratsg** [comment]
This also addresses #207.

**Pull request merged**